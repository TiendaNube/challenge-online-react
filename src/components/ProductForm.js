import React, { useState } from 'react';

import './components.scss';

const ProductForm = () => {
  const [product, setProduct] = useState({
    name: '',
    count: '',
    price: '',
    promotionalPrice: '',
  });

  const handleInputChange = (event) => {
    return setProduct(event.target.value);
  }

  const handleSubmit = (event) => {
    event.preventDefault();
  }

  return (
    <form className="centered" onSubmit={handleSubmit}>
      <h2>Add new product</h2>
      <div>
        <label className="label" htmlFor="name">Product name</label>
        <input id="name" name="name" type="text" value={ product.name } onChange={handleInputChange}/>
        <label className="label" htmlFor="count">Items count</label>
        <input id="count" name="count" type="text" value={ product.count } onChange={handleInputChange}/>
        <label className="label" htmlFor="price">Price</label>
        <input id="price" name="price" type="text" value={ product.price } onChange={handleInputChange}/>
        <label className="label" htmlFor="promotionalPrice">Promotional Price</label>
        <input id="promotionalPrice" name="promotionalPrice" type="text" value={ product.promotionalPrice } onChange={handleInputChange}/>
        <input className="button" type="submit" value="Aceptar" />
      </div>
    </form>
  );
}

export default ProductForm;
