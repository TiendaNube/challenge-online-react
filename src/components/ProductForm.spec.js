import '@testing-library/jest-dom/extend-expect';

import React from "react";
import { render, screen } from '@testing-library/react';
import ProductForm from "./ProductForm";

describe('ProductForm', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = render(<ProductForm />);
    });

    it('renders correctly', () => {
        const { container } = wrapper;
        expect(container).toMatchSnapshot();
    });

    describe('accessibility interactions', () => {
        it('should find product name input by its label', () => {
            const inputElement = screen.getByLabelText('Product name');
            expect(inputElement).not.toBeUndefined();
        });

        it('should find product count input by its label', () => {
            const inputElement = screen.getByLabelText('Items count');
            expect(inputElement).not.toBeUndefined();
        });

        it('should find product price input by its label', () => {
            const inputElement = screen.getByLabelText('Price');
            expect(inputElement).not.toBeUndefined();
        });

        it('should find product promotionalPrice input by its label', () => {
            const inputElement = screen.getByLabelText('Promotional Price');
            expect(inputElement).not.toBeUndefined();
        });
    });
});