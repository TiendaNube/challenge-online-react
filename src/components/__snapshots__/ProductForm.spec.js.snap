// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductForm renders correctly 1`] = `
<div>
  <form
    class="centered"
  >
    <h2>
      Add new product
    </h2>
    <div>
      <label
        class="label"
        for="name"
      >
        Product name
      </label>
      <input
        id="name"
        name="name"
        type="text"
        value=""
      />
      <label
        class="label"
        for="count"
      >
        Items count
      </label>
      <input
        id="count"
        name="count"
        type="text"
        value=""
      />
      <label
        class="label"
        for="price"
      >
        Price
      </label>
      <input
        id="price"
        name="price"
        type="text"
        value=""
      />
      <label
        class="label"
        for="promotionalPrice"
      >
        Promotional Price
      </label>
      <input
        id="promotionalPrice"
        name="promotionalPrice"
        type="text"
        value=""
      />
      <input
        class="button"
        type="submit"
        value="Aceptar"
      />
    </div>
  </form>
</div>
`;
